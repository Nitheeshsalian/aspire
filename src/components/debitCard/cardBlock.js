import React, {useState} from 'react';
import {View, Image, Switch, TouchableOpacity} from 'react-native';
import Text from '../../utils/text';
import styles from './debitStyle';

//import images
const hideEye = require('../../assets/images/Group.png');
const aspireLogo = require('../../assets/images/AspireLogo.png');
const visaLogo = require('../../assets/images/VisaLogo.png');
const remove_red_eye = require('../../assets/images/remove_red_eye.png');

export function CardBlock() {
  const [hideCard, setHideCard] = useState(false);

  const toggleCard = () => {
    setHideCard(!hideCard);
  };
  return (
    <React.Fragment>
      <View style={styles.greenCard}>
        <View style={styles.aspireLogoRow}>
          <Image style={styles.aspireLogo} source={aspireLogo} />
        </View>
        <Text style={styles.nameText}>Mark Henry</Text>
        <Text style={styles.cardNumberText}>
          {hideCard
            ? '5 6 4 7    3 4 1 1    2 4 1 3    2 0 2 0'
            : '* * * *    * * * *    * * * *    2 0 2 0'}
        </Text>
        <Text style={styles.validityText}>
          Thru: 12/20 CVV:{hideCard ? ' 4 5 6' : ' * * *'}
        </Text>
        <View style={styles.visaRow}>
          <Image style={styles.visaLogo} source={visaLogo} />
        </View>
      </View>
      <TouchableOpacity style={styles.hideCardContent} onPress={toggleCard}>
        <Image
          style={styles.eyeImage}
          source={hideCard ? hideEye : remove_red_eye}
        />
        <Text style={styles.hideText}>
          {hideCard ? 'Hide card number' : 'Show card number'}
        </Text>
      </TouchableOpacity>
    </React.Fragment>
  );
}
