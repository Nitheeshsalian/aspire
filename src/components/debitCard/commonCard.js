import React, {useState} from 'react';
import {View, Image, Switch} from 'react-native';
import Text from '../../utils/text';
import styles from './debitStyle';

// Common card component which takes image source, header text and body as Parameters
export function CommonCard({
  navigation,
  source,
  header,
  body,
  isSwitch = false,
}) {
  const [isEnabled, setIsEnabled] = useState(false);
  const toggleSwitch = () => {
    setIsEnabled(previousState => !previousState);
    navigation.navigate('SpendingLimit');
  };

  return (
    <View style={styles.textBox}>
      <View style={styles.row}>
        <Image style={styles.imageH12} source={source} />
        <View style={styles.pl12}>
          <Text style={styles.topText}>{header}</Text>
          <Text style={styles.bottomText}>{body}</Text>
        </View>
      </View>
      {isSwitch ? (
        <Switch
          trackColor={{false: '#EEEEEE', true: '#01D167'}}
          style={{transform: [{scaleX: 0.6}, {scaleY: 0.6}], marginRight: -10}}
          thumbColor="#f4f3f4"
          ios_backgroundColor="#EEEEEE"
          onValueChange={toggleSwitch}
          value={isEnabled}
        />
      ) : null}
    </View>
  );
}
