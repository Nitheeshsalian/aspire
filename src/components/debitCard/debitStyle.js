import React from 'react';
import { View, Image, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    container: { flex: 1, paddingLeft: 25, paddingRight: 25, paddingTop: 50, backgroundColor: '#0C365A' },
    headerWrapper: { flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between' },
    logo: {
        width: 25,
        height: 25,
    },
    headerText: {
        fontSize: 24,
        fontWeight: 'bold',
        paddingTop: 20,
    },
    availableText: {
        fontSize: 14,
        fontWeight: '500',
    },
    row: {
        flexDirection: 'row'
    },
    balanceSqaure: {
        paddingLeft: 4,
        paddingRight: 4,
        paddingTop: 4,
        padddingBottom: 4,
        backgroundColor: '#01D167',
        borderRadius: 5,
        width: 40,
        height: 24,
        alignItems: 'center',
        marginTop: 15
    },
    balanceWrapper: { flexDirection: 'row', alignItems: 'center' },
    s$: { fontWeight: 'bold', fontSize: 12 },
    balanceText: { fontSize: 24, marginTop: 15, marginLeft: 10, fontWeight: 'bold' },
    cardWrapper: { position: 'absolute', top: -100, backgroundColor: '#fff', width: '100%', borderTopStartRadius: 20, borderTopRightRadius: 20, height: '200%' },
    hideCardContent: {
        backgroundColor: '#ffff', width: 150, borderTopStartRadius: 10,
        borderTopRightRadius: 10, height: 44, position: 'absolute',
        top: -80, right: 20, zIndex: 0, alignItems: 'center',
        flexDirection: 'row', justifyContent: 'center',
    },
    eyeImage: { height: 16, width: 16, marginBottom: 10 },
    hideText: { color: '#01D167', paddingLeft: 5, marginBottom: 10, fontFamily: 'Avenir Next', fontWeight: '600', fontSize: 12 },
    greenCard: { backgroundColor: '#01D167', margin: 20, borderRadius: 10, height: 220, marginTop: -50, zIndex: 1 },
    aspireLogoRow: { flexDirection: 'row', alignItems: 'center', justifyContent: 'flex-end', paddingRight: 24, paddingTop: 24 },
    aspireLogo: { height: 21, width: 74 },
    nameText: {
        fontSize: 22, fontWeight: 'bold', paddingLeft: 24, paddingTop: 24
    },
    cardNumberText: {
        fontSize: 14, fontWeight: '600', paddingLeft: 24, paddingTop: 24
    },
    validityText: {
        fontSize: 14, fontWeight: '600', paddingLeft: 24, paddingTop: 15
    },
    visaLogo: { height: 20, width: 60 },
    visaRow: { flexDirection: 'row', alignItems: 'center', justifyContent: 'flex-end', paddingRight: 24 },
    imageH12: {
        height: 32,
        width: 32
    },
    pl12: { paddingLeft: 12 },
    topText: { color: '#000', fontSize: 14, color: '#25345F' },
    bottomText: { color: '#000', fontSize: 13, color: '#222222', fontWeight: '200' },
    textBox: { flexDirection: 'row', marginTop: 32, justifyContent: 'space-between' },
    spendingLimitHeader: {
        fontSize: 24,
        fontWeight: 'bold',
    },
    spendingContainer: { flex: 1,paddingTop: 50, backgroundColor: '#0C365A' },
    headerWrapper2: { marginLeft: 25, marginRight: 25, flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between'  },
    spendingWrapper:{ position: 'absolute', top: 153, backgroundColor: '#fff', width: '100%', borderTopStartRadius: 20, borderTopRightRadius: 20, height: '86%' ,justifyContent: 'space-between', paddingLeft: 24, paddingRight: 24},
    spendingHeader:{ marginTop: 24, paddingLeft:30 },
    balanceSqaure2: {
        paddingLeft: 4,
        paddingRight: 4,
        paddingTop: 4,
        padddingBottom: 4,
        backgroundColor: '#01D167',
        borderRadius: 5,
        width: 40,
        height: 24,
        alignItems: 'center',
        marginTop: 10
    },
    amountText: {
        backgroundColor: '#effcf4',
        height: 38,
        width: 94, alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 5,
        borderColor: '#20D167'
    },
    amountTextStyle: { color: '#01D167', fontWeight: '700' },
    inputContainer: {
        flexDirection: 'row', borderBottomColor: '#22222266', borderBottomWidth: 1, marginTop: 10,
        alignContent: 'center', height: 44, justifyContent: 'flex-start'
    },
    inputAmount:{ width: '100%', height: 44, marginLeft: 10, fontWeight: 'bold' },
    infoText:{ color: "#22222266", fontSize: 12, marginTop: 12 },
    amountGroup: {
        marginTop: 32,
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    saveButton: {
        backgroundColor: '#01D167',
        height: 56,
        width: 300,
        marginLeft: 20,
        borderRadius: 50,
        justifyContent:'center',
        alignItems: 'center',
        marginBottom: 20
    },
    saveText: { fontSize: 16, fontWeight: '700'},
    
})

export default styles;