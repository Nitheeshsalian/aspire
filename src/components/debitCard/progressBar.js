import React, {useState, useRef, useEffect} from 'react';
import {View} from 'react-native';
import Text from '../../utils/text';
import styles from './progressStyle';
//reducer import
import {useSelector} from 'react-redux';
const debitStore = state => state.debit;

export const ProgressBar = () => {
  const [getPercentage, setPercentage] = useState('20%');
  const debit = useSelector(debitStore);

  useEffect(() => {
    if (debit.limit > 0) {
      let per = debit.current / debit.limit;
      setPercentage(per * 100 + '%');
    }
  }, [debit]);

  return (
    <View>
      <View style={styles.debitCardWrap}>
        <Text style={styles.debitCardText}>Debit card spending limit</Text>
        <View style={styles.flexRow}>
          <Text style={styles.spentText}>
            ${debit.current.toLocaleString('en-US')}
          </Text>
          <Text style={styles.remainingText}>
            {' '}
            | ${debit.limit.toLocaleString('en-US')}
          </Text>
        </View>
      </View>
      <View style={styles.progressBarContain}>
        <View style={styles.progressBarBackground} />
        <View
          style={[
            {
              width: getPercentage ? getPercentage : 0,
            },
            styles.progressBar,
          ]}
        />
      </View>
    </View>
  );
};
