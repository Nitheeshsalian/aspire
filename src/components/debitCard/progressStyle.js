import React from 'react';
import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  debitCardWrap: {flexDirection: 'row', justifyContent: 'space-between'},
  debitCardText: {color: '#222222', fontSize: 13, fontWeight: '500'},
  flexRow: {flexDirection: 'row'},
  spentText: {color: '#01D167', fontSize: 13, fontWeight: '500'},
  remainingText: {color: '#22222233', fontSize: 13, fontWeight: '500'},
  progressBarContain: {justifyContent: 'center', marginTop: 6},
  progressBarBackground: {
    width: '100%',
    height: 15,
    marginVertical: 10,
    borderRadius: 20,
    borderColor: '#cceddd',
    borderWidth: 1,
    backgroundColor: '#cceddd',
  },
  progressBar: {
    height: 15,
    marginVertical: 10,
    borderTopLeftRadius: 15,
    borderBottomLeftRadius: 15,
    backgroundColor: '#01D167',
    position: 'absolute',
    bottom: 20,
    borderStyle: 'solid',
    borderRightWidth: 15,
    borderTopWidth: 15,
    borderRightColor: 'transparent',
    borderTopColor: '#01D167',
    top: 0,
  },
});

export default styles;
