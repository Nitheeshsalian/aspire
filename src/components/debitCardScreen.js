import * as React from 'react';
import { View, Image, ScrollView } from 'react-native';
import styles from './debitStyle';
import Text from '../utils/text'; //Custom text component where default font style have been set.
const logo = require('../assets/images/Logo.png');
const hideEye = require('../assets/images/Group.png');
const aspireLogo = require('../assets/images/AspireLogo.png');
const visaLogo = require('../assets/images/VisaLogo.png');

export function DebitCardScreen() {
  return (
    <React.Fragment>

      <View style={styles.container}>
        <View style={styles.headerWrapper}>
          <Text style={styles.headerText}>Debit Card</Text>
          <Image
            style={styles.logo}
            source={logo}
          />
        </View>
        <View style={{ marginTop: 24 }}>
          <Text styles={styles.availableText}>Available balance</Text>
          <View style={styles.balanceWrapper}>
            <View style={styles.balanceSqaure}>
              <Text style={styles.s$}>S$</Text>
            </View>
            <Text style={styles.balanceText}>3,000</Text>
          </View>
        </View>
      </View>

      <View style={styles.cardWrapper}>
        <View style={styles.greenCard}>
          <View style={styles.aspireLogoRow}>
            <Image
              style={styles.aspireLogo}
              source={aspireLogo}
            />
          </View>
          <Text style={styles.nameText}>Mark Henry</Text>

          <Text style={styles.cardNumberText}>5 6 4 7    3 4 1 1    2 4 1 3    2 0 2 0</Text>

          <Text style={styles.validityText}>Thru: 12/20     CVV: 456</Text>
          <View style={styles.visaRow}>
            <Image
              style={styles.visaLogo}
              source={visaLogo}
            />
          </View>
        </View>
        <View style={styles.hideCardContent}>
          <Image
            style={styles.eyeImage}
            source={hideEye}
          />
          <Text style={styles.hideText}>Hide card number</Text>
        </View>
        <Text>
          nojfdmnvkjdfmvjkdfm
        </Text>
      </View>
    </React.Fragment>
  );
}
