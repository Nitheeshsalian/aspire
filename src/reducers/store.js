import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import rootReducer from './index';
import debitSaga from '../sagas/debitSaga';

// create the saga middleware
const sagaMiddleware = createSagaMiddleware();
// mount it on the Store
const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));

// then run the saga
sagaMiddleware.run(debitSaga);

// render the application

export default store;
