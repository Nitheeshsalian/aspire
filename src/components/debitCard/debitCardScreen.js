import React, { useState, useRef } from 'react';
import { View, Image, Switch, Animated, PanResponder } from 'react-native';
import styles from './debitStyle';
//Custom text component where default font style have been set.
import Text from '../../utils/text';
import { CommonCard } from './commonCard';
import { CardBlock } from './cardBlock';

//import images
const logo = require('../../assets/images/Logo.png');
const insightLogo = require('../../assets/images/insight.png');
const transfer = require('../../assets/images/Transfer.png');
const freeze = require('../../assets/images/Freeze.png');
const deactivated = require('../../assets/images/deactivated.png');


export function DebitCardScreen() {

  //pan responder to move debit card vertically
  //This function needs to be fixed to move only sepcified height 
  const pan = useRef(new Animated.ValueXY()).current;
  const panResponder = useRef(
    PanResponder.create({
      onMoveShouldSetPanResponder: () => true,
      onPanResponderGrant: () => {
        pan.setOffset({
          x: pan.x._value,
          y: pan.y._value
        });
      },
      onPanResponderMove: Animated.event(
        [
          null,
          { dy: pan.y },
        ],
        { useNativeDriver: false }
      ),
      onPanResponderRelease: () => {
        pan.flattenOffset();
      }
    })
  ).current;


  return (
    <React.Fragment>

      <View style={styles.container}>
        <View style={styles.headerWrapper}>
          <Text style={styles.headerText}>Debit Card</Text>
          <Image
            style={styles.logo}
            source={logo}
          />
        </View>
        <View style={{ marginTop: 24 }}>
          <Text styles={styles.availableText}>Available balance</Text>
          <View style={styles.balanceWrapper}>
            <View style={styles.balanceSqaure}>
              <Text style={styles.s$}>S$</Text>
            </View>
            <Text style={styles.balanceText}>3,000</Text>
          </View>
        </View>
      </View>
      <Animated.View
        style={{
          transform: [{ translateY: pan.y }],
          flex: 1
        }}
        {...panResponder.panHandlers}

      >
        <View style={styles.cardWrapper}>

          <CardBlock />
          {/* Card menu options */}
          <View style={{ paddingTop: 0, marginLeft: 32, marginRight: 32 }}>
            <CommonCard
              source={insightLogo}
              header="Top-up account"
              body="Deposit money to your account to use with card"
            />

            <CommonCard
              source={transfer}
              header="Weekly spending limit"
              body="You haven't set any spending limit on card"
              isSwitch={true}
            />

            <CommonCard
              source={freeze}
              header="Freeze Card"
              body="Your debit card is currently active"
              isSwitch={true}
            />
            <CommonCard
              source={insightLogo}
              header="Get a new card"
              body="This deactivates your current card"
            />
            <CommonCard
              source={deactivated}
              header="Deactivated card"
              body="Your previously deactivated cards"
            />
          </View>

        </View>
      </Animated.View>
    </React.Fragment>
  );
}
