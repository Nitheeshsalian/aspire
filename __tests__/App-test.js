/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import store from '../src/reducers/store'
import { Provider } from 'react-redux'
import Routes from '../src/index';
import { DebitCardScreen } from '../src/components/debitCard/debitCardScreen'

//Mock font awesome
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: ''
}))

describe('<App>', () => {
  it('renders app correctly', () => {
    renderer.create(
      <Provider store={store}>
        <Routes />
      </Provider>
    );
  });
});

//test debit card screen
describe('<DebitCardScreen>', () => {
  it('renders DebitCardScreen correctly', () => {
    renderer.create(
      <Provider store={store}>
        <DebitCardScreen />
      </Provider>
    );
  });


  //create snapshot and compare
  it('test snapshot', () => {
    const tree = renderer
      .create(<Provider store={store}>
        <DebitCardScreen />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});


describe("Reducer test", () => {

  it("Should set the supplied initial state", () => {
    expect(store.getState()).toEqual({
      debit: {
        limit: 1000
      }
    });
  });

  it("Mock update action set limit", () => {
    const dispatch = jest.fn();
    expect(store.getState()).toEqual({
      debit: {
        limit: 1000
      }
    });
  });
});