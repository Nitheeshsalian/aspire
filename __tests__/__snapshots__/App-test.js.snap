// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DebitCardScreen> test snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#0C365A",
        "flex": 1,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 50,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir Next",
            },
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "paddingTop": 20,
            },
          ]
        }
      >
        Debit Card
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginTop": 24,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir Next",
            },
            undefined,
          ]
        }
        styles={
          Object {
            "fontSize": 14,
            "fontWeight": "500",
          }
        }
      >
        Available balance
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#01D167",
              "borderRadius": 5,
              "height": 24,
              "marginTop": 15,
              "padddingBottom": 4,
              "paddingLeft": 4,
              "paddingRight": 4,
              "paddingTop": 4,
              "width": 40,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir Next",
                },
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            S$
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir Next",
              },
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginTop": 15,
              },
            ]
          }
        >
          1000
        </Text>
      </View>
    </View>
  </View>,
  <View
    collapsable={false}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "flex": 1,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopRightRadius": 20,
          "borderTopStartRadius": 20,
          "height": "200%",
          "position": "absolute",
          "top": -100,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#01D167",
            "borderRadius": 10,
            "height": 220,
            "margin": 20,
            "marginTop": -50,
            "zIndex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "paddingRight": 24,
              "paddingTop": 24,
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 21,
                "width": 74,
              }
            }
          />
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir Next",
              },
              Object {
                "fontSize": 22,
                "fontWeight": "bold",
                "paddingLeft": 24,
                "paddingTop": 24,
              },
            ]
          }
        >
          Mark Henry
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir Next",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "600",
                "paddingLeft": 24,
                "paddingTop": 24,
              },
            ]
          }
        >
          * * * *    * * * *    * * * *    2 0 2 0
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir Next",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "600",
                "paddingLeft": 24,
                "paddingTop": 15,
              },
            ]
          }
        >
          Thru: 12/20     CVV:
           * * *
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "paddingRight": 24,
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 20,
                "width": 60,
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffff",
            "borderTopRightRadius": 10,
            "borderTopStartRadius": 10,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "right": 20,
            "top": -80,
            "width": 150,
            "zIndex": 0,
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 16,
              "marginBottom": 10,
              "width": 16,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir Next",
              },
              Object {
                "color": "#01D167",
                "fontFamily": "Avenir Next",
                "fontSize": 12,
                "fontWeight": "600",
                "marginBottom": 10,
                "paddingLeft": 5,
              },
            ]
          }
        >
          Show card number
        </Text>
      </View>
      <View
        style={
          Object {
            "marginLeft": 32,
            "marginRight": 32,
            "paddingTop": 0,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 32,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#25345F",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Top-up account
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#222222",
                      "fontSize": 13,
                      "fontWeight": "200",
                    },
                  ]
                }
              >
                Deposit money to your account to use with card
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 32,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#25345F",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Weekly spending limit
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#222222",
                      "fontSize": 13,
                      "fontWeight": "200",
                    },
                  ]
                }
              >
                You haven't set any spending limit on card
              </Text>
            </View>
          </View>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#01D167"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Array [
                  Object {
                    "marginRight": -10,
                    "transform": Array [
                      Object {
                        "scaleX": 0.6,
                      },
                      Object {
                        "scaleY": 0.6,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "#EEEEEE",
                    "borderRadius": 16,
                  },
                ],
              ]
            }
            thumbTintColor="#f4f3f4"
            tintColor="#EEEEEE"
            value={false}
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 32,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#25345F",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Freeze Card
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#222222",
                      "fontSize": 13,
                      "fontWeight": "200",
                    },
                  ]
                }
              >
                Your debit card is currently active
              </Text>
            </View>
          </View>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#01D167"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Array [
                  Object {
                    "marginRight": -10,
                    "transform": Array [
                      Object {
                        "scaleX": 0.6,
                      },
                      Object {
                        "scaleY": 0.6,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "#EEEEEE",
                    "borderRadius": 16,
                  },
                ],
              ]
            }
            thumbTintColor="#f4f3f4"
            tintColor="#EEEEEE"
            value={false}
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 32,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#25345F",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Get a new card
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#222222",
                      "fontSize": 13,
                      "fontWeight": "200",
                    },
                  ]
                }
              >
                This deactivates your current card
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 32,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#25345F",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Deactivated card
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Avenir Next",
                    },
                    Object {
                      "color": "#222222",
                      "fontSize": 13,
                      "fontWeight": "200",
                    },
                  ]
                }
              >
                Your previously deactivated cards
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
